tests/functional/test_vcs_mercurial.py

```diff
@@ -0,0 +1,29 @@
+"""
+Contains functional tests of the Mercurial class.
+"""
+
+import os
+
+from pip._vendor.six.moves import configparser
+
+from pip._internal.utils.temp_dir import TempDirectory
+from pip._internal.vcs import RevOptions
+from pip._internal.vcs.mercurial import Mercurial
+
+
+def test_mercurial_switch_updates_config_file_when_found():
+    hg = Mercurial()
+    options = RevOptions(hg)
+    with TempDirectory() as temp:
```

Comment:
I believe there is a pytest construct to avoid explicitly using `TempDirectory` (consult other tests to find the pattern).

Reply:
I cannot find anything that wraps TempDirectory, any hint on which test may hve the pattern?

Reply:
Look for tests that have `tmpdir` in their signature. I believe pytest passes the temp directory automatically for such tests, and cleans up.

Reply:
Found, thanks! updated tests to use tmpdir.